File: core/commands/registry/expect/send_tty.py
"""send_tty -- Send a string to the controlling tty."""

from __future__ import annotations

from .._base import CommandDef
from ..models import CommandSpec, FormKind, FormSpec, HoverSnippet, OptionSpec, ValidationSpec
from ..signatures import Arity
from ._base import _EXPECT_ONLY, register

_SOURCE = "Expect send_tty(1)"


@register
class SendTtyCommand(CommandDef):
    name = "send_tty"

    @classmethod
    def spec(cls) -> CommandSpec:
        return CommandSpec(
            name="send_tty",
            dialects=_EXPECT_ONLY,
            hover=HoverSnippet(
                summary="Send a string to the controlling terminal (tty).",
                synopsis=("send_tty ?-flags? string",),
                source=_SOURCE,
            ),
            forms=(
                FormSpec(
                    kind=FormKind.DEFAULT,
                    synopsis="send_tty ?-flags? string",
                    options=(
                        OptionSpec(name="-raw", detail="Send without translation."),
                        OptionSpec(name="--", detail="End of options."),
                    ),
                ),
            ),
            validation=ValidationSpec(arity=Arity(1)),
        )

File: rust/tcl-registry/src/commands/expect/send_tty.rs
//! `send_tty` command.
use crate::prelude::*;
const OPTIONS: &[OptionSpec] = &[
    OptionSpec {
        name: "-raw",
        takes_value: false,
        value_hint: "",
        detail: "Send without translation.",
        dialects: None,
    },
    OptionSpec {
        name: "--",
        takes_value: false,
        value_hint: "",
        detail: "End of options.",
        dialects: None,
    },
];

const FORMS: &[FormSpec] = &[FormSpec {
    kind: FormKind::Default,
    synopsis: "send_tty ?-flags? string",
}];

pub fn spec() -> CommandSpec {
    CommandSpec {
        name: "send_tty",
        dialects: Some(DialectSet::EXPECT),
        arity: Arity::at_least(1),
        hover: Some(HoverSnippet {
            summary: "Send a string to the controlling terminal (tty).",
            synopsis: &["send_tty ?-flags? string"],
            snippet: "",
            source: "Expect send_tty(1)",
            examples: "",
            return_value: "",
        }),
        forms: FORMS,
        options: OPTIONS,
        ..CommandSpec::DEFAULT
    }
}
